wrappers/wrapper.rb

```diff
@@ -1111,6 +1113,139 @@ def simplify_vehicle_pause(vrp, result = nil, options = { mode: :simplify })
+            additional_setup = vehicle.additional_setup.to_i
+
+            # WARNING: Here we apply the setup_duration for the points which has non-zero
+            # distance (in time!) between them because this is the case in optimizer-ortools.
+            # If this is changed to per "destination" (i.e., point.id) based setup duration
+            # then the following logic needs to be updated. Basically we need to do each_with_index
+            # and apply the setup duration increment to every pair except index == point.matrix_index
+            # even if they were 0 in the first place.
+            matrix.time.each{ |row|
+              row[point.matrix_index] += (coef_setup * setup_duration + additional_setup).to_i if row[point.matrix_index] > 0
+            }
+          }
+        }
+
+        return nil unless vrp.services.any?{ |s| s[:simplified_setup_duration] }
+
+        vrp.vehicles.each{ |vehicle|
+          vehicle[:simplified_coef_setup] = vehicle.coef_setup
+          vehicle[:simplified_additional_setup] = vehicle.additional_setup
+          vehicle.coef_setup = nil
+          vehicle.additional_setup = nil
+        }
+      when :rewind
+        # take it back in case in dicho and there will be re-optimization
+        return nil unless vrp.services.any?{ |s| s[:simplified_setup_duration] }
+
+        vehicles_grouped_by_matrix_id = vrp.vehicles.group_by(&:matrix_id)
+
+        vrp.services.group_by{ |s| s.activity.point }.each{ |point, service_group|
+          setup_duration = service_group.first[:simplified_setup_duration].to_i
+
+          next if setup_duration.zero?
+
+          vrp.matrices.each{ |matrix|
+            vehicle = vehicles_grouped_by_matrix_id[matrix.id].first
+            coef_setup = vehicle[:simplified_coef_setup] || 1
+            additional_setup = vehicle[:simplified_additional_setup].to_i
+
+            matrix.time.each{ |row|
+              row[point.matrix_index] -= (coef_setup * setup_duration + additional_setup).to_i  if row[point.matrix_index] > 0
+            }
+          }
+
+          service_group.each{ |service|
+            service.setup_duration = service[:simplified_setup_duration]
+            service[:simplified_setup_duration] = nil
+          }
+        }
+
+        vrp.vehicles.each{ |vehicle|
+          vehicle.coef_setup = vehicle[:simplified_coef_setup]
+          vehicle.additional_setup = vehicle[:simplified_additional_setup]
+          vehicle[:simplified_coef_setup] = nil
+          vehicle[:simplified_additional_setup] = nil
+        }
+      when :patch_result
+        # patches the result
+        # the travel_times need to be decreased and setup_duration need to be increased by
+        # (coef_setup * setup_duration + additional_setup) if setup_duration > 0 and travel_time > 0
+        return nil unless vrp.services.any?{ |s| s[:simplified_setup_duration] }
+
+        vehicles_grouped_by_vehicle_id = vrp.vehicles.group_by(&:id)
```

Comment:
can we group by id ? I assume it is supposed to be unique. original_id may not be unique

I you are expecting to have exactly one vehicle by group, I would be expecting a hash id=>vehicle.